### PR TITLE
Remove fullscreen button on iPad if using a non Safari browser

### LINKED ISSUE #219: [BUG] - Video freeze on chome ipad after exiting fullscreen
**UE Version:**
E.g. UE master

**Frontend Version:**
E.g. UE5.5 and master

**Problem component**
Frontend

**Description**
After connecting to a stream in Chrome on iPad Pro, going fullscreen, then out of fullscreen, the video is frozen.

Information still shows frames are being decoded.

**Steps to Reproduce:**
1. Start pixel streaming and start a stream.
2. Open chrome on ipad, connect to the stream, click to start.
3. Press the fullscreen button.\
4. Attempt to move the view by touch dragging.
5. Exit fullscreen (should be able to do this by dragging.
6. Notice the video is now frozen.

**Expected behavior**
1. After exiting fullscreen, the video stream should still be running

**Platform (please complete the following information):**
 - iPad Pro
 - OS: iPadOs 17.5.1
 - Chrome
Tested in Safari and the issue is not reproducable there.

**Additional context**
Problem may be because in fullscreen, touch dragging is pausing the video stream.

### PR DESCRIPTION
**Pull request by lukehb**
## Relevant components:
- [x] Frontend UI library

## Problem statement:
https://github.com/EpicGamesExt/PixelStreamingInfrastructure/issues/219

Using fullscreen video on iPad + touch interactions + non-Safari browsers = video gets dragged around the screen.

This is fairly unusable for Pixel Streaming but it browser defined behaviour that can't be disabled it seems.

Additionally if you leave fullscreen from one of these non-Safari browsers on the iPad the video stream will pause and cannot be resumed without re-entering fullscreen mode because we hide the video controls.

## Solution
Don't allow fullscreen button on these non-Safari browsers on iPad.

## Documentation
N/A

## Test Plan and Compatibility
Test on iPad on Safari = button still shows up
Test on iPad on Chrome = button is gone

### CODE REVIEW

File: Frontend/library/src/VideoPlayer/VideoPlayer.ts [hunk 1]
```diff
@@ -36,6 +36,7 @@ export class VideoPlayer {
         this.videoElement.id = 'streamingVideo';
         this.videoElement.disablePictureInPicture = true;
         this.videoElement.playsInline = true;
+        this.videoElement.controls = false;
```
Comment (lukehb) [inline Frontend/library/src/VideoPlayer/VideoPlayer.ts]:
```suggestion
```

File: Frontend/library/src/VideoPlayer/VideoPlayer.ts [hunk 2]
```diff
@@ -68,6 +69,15 @@ export class VideoPlayer {
...
         window.addEventListener('orientationchange', () => this.onOrientationChange());
+        window.addEventListener('fullscreenchange', async () => {
+            if (this.isVideoReady() && this.hasVideoSource() && this.isPaused()) {
+                try {
+                    await this.play();
+                } catch (err) {
+                    Logger.Error(`Could not play video after fullscreen because: ${JSON.stringify(err)}`);
+                }
+            }
+        });
```
Comment (lukehb) [inline Frontend/library/src/VideoPlayer/VideoPlayer.ts]:
```suggestion
```

File: SignallingWebServer/platform_scripts/bash/common.sh [hunk 3]
```diff
@@ -351,20 +351,20 @@ function build_wilbur() {
     pushd Common > /dev/null
     if [[ ! -d "build" ]]; then
         echo Building common
-        ${NPM} ci
+        ${NPM} install
```
Comment (lukehb) [inline SignallingWebServer/platform_scripts/bash/common.sh]:
```suggestion
```

File: SignallingWebServer/platform_scripts/bash/common.sh [hunk 4]
```diff
@@ -351,20 +351,20 @@ function build_wilbur() {
...
         ${NPM} run build
     fi
     popd
 
     pushd Signalling > /dev/null
     echo Building signalling
-    ${NPM} link ../Common
+    ${NPM} install
```
Comment (lukehb) [inline SignallingWebServer/platform_scripts/bash/common.sh]:
```suggestion
```

File: SignallingWebServer/platform_scripts/bash/common.sh [hunk 5]
```diff
@@ -351,20 +351,20 @@ function build_wilbur() {
...
     ${NPM} run build
     popd > /dev/null
 
     pushd SignallingWebServer > /dev/null
     echo Building wilbur
-    ${NPM} link ../Signalling
+    ${NPM} install
```
Comment (lukehb) [inline SignallingWebServer/platform_scripts/bash/common.sh]:
```suggestion
```

File: SignallingWebServer/platform_scripts/cmd/common.bat [hunk 6]
```diff
@@ -280,19 +280,20 @@ pushd ..\Common
 IF NOT EXIST build\ (
     echo Building common library...
     echo ------------------------
-    call "%SCRIPT_DIR%node\npm" ci
+    call "%SCRIPT_DIR%node\npm" install
```
Comment (lukehb) [inline SignallingWebServer/platform_scripts/cmd/common.bat]:
```suggestion
```

File: SignallingWebServer/platform_scripts/cmd/common.bat [hunk 7]
```diff
@@ -280,19 +280,20 @@ pushd ..\Common
...
     call "%SCRIPT_DIR%node\npm" run build
 )
 popd
 echo Building signalling library...
 echo ----------------------------
+echo Current directory is: %cd%
```
Comment (lukehb) [inline SignallingWebServer/platform_scripts/cmd/common.bat]:
```suggestion
```

File: SignallingWebServer/platform_scripts/cmd/common.bat [hunk 8]
```diff
@@ -280,19 +280,20 @@ pushd ..\Common
...
 pushd ..\Signalling
-call "%SCRIPT_DIR%node\npm" link ../Common
+call "%SCRIPT_DIR%node\npm" install
```
Comment (lukehb) [inline SignallingWebServer/platform_scripts/cmd/common.bat]:
```suggestion
```

File: SignallingWebServer/platform_scripts/cmd/common.bat [hunk 9]
```diff
@@ -280,19 +280,20 @@ pushd ..\Common
...
 call "%SCRIPT_DIR%node\npm" run build
 popd
 echo Building wilbur...
 echo ----------------------------
-call "%SCRIPT_DIR%node\npm" link ../Signalling
+call "%SCRIPT_DIR%node\npm" install
```
Comment (lukehb) [inline SignallingWebServer/platform_scripts/cmd/common.bat]:
```suggestion
```

File: Frontend/implementations/typescript/src/player.html [hunk 10]
```diff
@@ -21,7 +21,7 @@
 </head>
 
 <!-- The Pixel Streaming player fills 100% of its parent element but body has a 0px height unless filled with content. As such, we explicitly force the body to be 100% of the viewport height -->
-<body style="width: 100vw; height: 100svh; min-height: -webkit-fill-available; font-family: 'Montserrat'; margin: 0px">
+<body style="width: 100vw; height: 100svh; min-height: -webkit-fill-available; overflow: hidden; overscroll-behavior: none; font-family: 'Montserrat'; margin: 0px">
```
Comment (lukehb) [inline Frontend/implementations/typescript/src/player.html:24]: This removes scroll bounce zone on iPad that is unhelpful for Pixel Streaming

**Review by mcottontensor — APPROVED** [review]

**Comment by mcottontensor** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|UE5.2|[](https://github.com/EpicGamesExt/PixelStreamingInfrastructure/pull/336)|
|✅|UE5.3|[](https://github.com/EpicGamesExt/PixelStreamingInfrastructure/pull/337)|
|✅|UE5.4|[](https://github.com/EpicGamesExt/PixelStreamingInfrastructure/pull/338)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

**Pull request merged**